apsuite/loco/main.py

```diff
@@ -200,41 +203,61 @@ def _handle_quad_fit_k(self, fname_jloco_k_quad):
                     jloco_k_quad_dict, 'quadrupole')
 
     def _convert_dict2array(self, jlocodict, magtype, is_normal=True):
-        jloco = []
+
+        if 'knobs_order' in jlocodict:
+            knobs_order = jlocodict['knobs_order']
+        else:
+            raise Exception('knobs_order missing in jloco dictionary!')
         if magtype == 'dipole':
             if self.config.dipoles_to_fit is not None:
                 magstofit = self.config.dipoles_to_fit
             else:
                 magstofit = self.config.famname_dipset
+            index = self.config.dip_indices
         elif magtype == 'quadrupole':
             if self.config.quadrupoles_to_fit is not None:
                 magstofit = self.config.quadrupoles_to_fit
             else:
                 magstofit = self.config.famname_quadset
+            index = self.config.quad_indices
         elif magtype == 'sextupole':
             if self.config.sextupoles_to_fit is not None:
                 magstofit = self.config.sextupoles_to_fit
             else:
                 magstofit = self.config.famname_sextset
+            index = self.config.sext_indices
         elif magtype == 'skew_quadrupole':
-            magstofit = self.config.famname_skewquadset
+            if self.config.skew_quadrupoles_to_fit is not None:
+                magstofit = self.config.skew_quadrupoles_to_fit
+            else:
+                magstofit = self.config.famname_skewquadset
+            index = self.config.skew_quad_indices
         quadfam = self.config.use_quad_families
         quadfam &= magtype == 'quadrupole'
         quadfam &= is_normal
         dipfam = self.config.use_dip_families
         dipfam &= magtype == 'dipole'
         dipfam &= is_normal
         if dipfam or quadfam:
+            jloco = []
             for quad in magstofit:
                 famcols = [
```

Comment:
I think it's `famcoils` here, right?

Reply:
famcols is a short for family columns, the matrix columns that belong to the same family.

Reply:
kkk, my fault!